microsoft/react-native-test-app
# feat: add support for bridgeless mode

### Description

To enable bridgeless mode:

- **Android:** In `gradle.properties`:
  ```gradle
  newArchEnabled=true
  bridgelessEnabled=true
  ```

- **iOS:** In `Podfile`:
  ```ruby
  options = {
    :bridgeless_enabled => true,
    :fabric_enabled => true,
    :hermes_enabled => false,
  }

  use_test_app! options
  ```

See the full announcement here:
https://reactnative.dev/blog/2023/12/06/0.73-debugging-improvements-stable-symlinks#new-architecture-updates

Resolves #1723

### Platforms affected


### Test plan

1. In a separate terminal, start Metro server and keep it running for the rest of the session
   ```
   cd example
   yarn start
   ```
2. Android: Test old architecture
   ```
   yarn android
   ```
3. Android: Test new architecture
    - In `android/gradle.properties`, uncomment `newArchEnabled=true`
    - Repeat commands in step 2
    - Verify that "Fabric" is on
4. Android: Test new architecture + bridgeless
    - In `android/gradle.properties`, uncomment `bridgelessEnabled=true`
    - Repeat commands in step 2
    - Verify that "Bridgeless" is on
5. iOS: Test old architecture
   ```
   pod install --project-directory=ios
   yarn ios
   ```
6. iOS: Test new architecture
    - In `ios/Podfile`, change `:fabric_enabled` to `true`
    - Repeat commands in step 5
    - Verify that "Fabric" is on
7. iOS: Test new architecture + bridgeless
    - In `ios/Podfile`, change `:bridgeless_enabled` to `true`
    - Repeat commands in step 5
    - Verify that "Bridgeless" is on
8. iOS: Test Hermes
    - In `ios/Podfile`, change `:hermes_enabled` to `true`
    - Repeat commands in step 5
    - Verify that "Hermes" is on

### Screenshots

| Android | iOS + New Architecture | + Bridgeless | + Hermes |
| :-: | :-: | :-: | :-: |
| ![Screenshot_1706126818](https://github.com/microsoft/react-native-test-app/assets/4123478/ee26774f-cb2d-42f9-89db-f25c63e24848) | ![Simulator Screenshot - iPhone 15 Pro - 2024-01-25 at 09 45 06](https://github.com/microsoft/rnx-kit/assets/4123478/a704cc13-70d5-44c7-9e1d-bae89d26c3f2) | ![Simulator Screenshot - iPhone 15 Pro - 2024-01-24 at 21 07 40](https://github.com/microsoft/rnx-kit/assets/4123478/cca45e09-1321-4bcf-b73a-b21f0317248a) | ![Simulator Screenshot - iPhone 15 Pro - 2024-01-25 at 09 32 59](https://github.com/microsoft/rnx-kit/assets/4123478/89649ca5-7b00-44d3-bb6a-da32457462b0) |
